maksim_khier/handlers/user_handlers.go

```diff
@@ -0,0 +1,106 @@
+package handlers
+
+import (
+  "github.com/khier996/test-backend/maksim_khier/models"
+  "github.com/gin-gonic/gin"
+  "time"
+  "strconv"
+)
+
+func GetUser(c *gin.Context) {
+  userId, err := strconv.ParseUint(c.Param("id"), 10, 64)
+  if err != nil {
+    c.JSON(400, gin.H{"error": "user id is incorrect"})
+    return
+  }
+
+  var user models.User
+  wiredDB.First(&user, userId)
+  if user.ID != 0 {
+    c.JSON(200, user)
+  } else {
+    c.Status(404)
+  }
+}
+
+func GetUsers(c *gin.Context) {
+  offset := 0
+  limit := 10
+
+  if c.Query("per_page") != "" {
+    if per_page, err := strconv.ParseInt(c.Query("per_page"), 10 ,32); err == nil {
+      limit = int(per_page)
+    }
+  }
+  
+  if page, err := strconv.ParseInt(c.Query("page"), 10, 32); err == nil {
+    offset = limit * (int(page) - 1)
+  }  
+
+  var users []models.User
+  if getError := wiredDB.Offset(offset).Limit(limit).Find(&users).Error; getError != nil {
+    c.JSON(500, gin.H{"error": getError})
+  } else {
+    c.JSON(200, users)
+  }
+}
+
+func CreateUser(c *gin.Context) {
+  var user models.User
+  if err := c.ShouldBindJSON(&user); err != nil {
+    c.JSON(400, gin.H{"error": err.Error()})
+  } else {
+    if valid, errMes := user.Valid(); !valid { 
+      c.JSON(400, gin.H{"error": errMes})
+      return
+    } 
+    user.CreatedAt = time.Now()
+    if createErr := wiredDB.Create(&user).Error; createErr != nil {
+      c.JSON(400, gin.H{"error": createErr})
```

Comment:
Why do you think `createErr` would be a 400(bad request)?

Reply:
Since there is already a User.Validate method that checks for the validity of a user, the checking of createErr is just an extra precaution. If some error happens on the database level, we need to inform the client. Now that I think about the exact status code, I am not sure 400 is the best choice. Maybe 500 would be better.